qiskit/algorithms/amplitude_amplifiers/grover.py

```diff
@@ -342,7 +423,6 @@ def __init__(self) -> None:
         super().__init__()
         self._iterations = None
         self._circuit_results = None
-        self._shots = None
```

Comment:
I have to admit the GroverResult is kinda odd with respect to its parent. The parent has a getter/setter for _circuit_results but does not define it there in its constructor. I would argue the `self._circuit_results = None` should be moved to the AmplitudeAmplifierResult. It has existed like this for a while and though the code works, since the base class, like any Python, can define fields on the fly, it seems this could be corrected easily enough and the code would still work just the same. All I would expect to see here is self._iterations since that is additional over the parent